app/src/main/java/com/github/se/signify/ui/common/Practice.kt

```diff
@@ -0,0 +1,126 @@
+import androidx.compose.material.icons.automirrored.outlined.ArrowForward
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.signify.model.common.getIconResId
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * Composable function that displays a horizontally arranged letter navigator. Users can scroll
+ * through each letter in the alphabet using left and right arrow buttons, and click on a letter box
+ * to scroll to the corresponding item in a vertically scrollable list.
+ *
+ * @param coroutineScope The `CoroutineScope` for launching scroll actions.
+ * @param numbOfHeaders The number of headers at the top of the list, allowing for an offset when
+ *   scrolling to the selected letter.
+ */
+@Composable
+fun LetterDictionary(
+    coroutineScope: CoroutineScope,
+    numbOfHeaders: Int,
+    onClick: (Int, Int) -> Unit
+) {
+  val letters = ('a'..'z').toList()
+  val pagerState = rememberPagerState(initialPage = 0, pageCount = { letters.size })
+
+  Box(modifier = Modifier.fillMaxWidth().wrapContentHeight().testTag("LetterDictionary")) {
+    // Back Arrow at the start
+    IconButton(
+        onClick = {
+          coroutineScope.launch {
+            pagerState.animateScrollToPage(
+                (pagerState.currentPage - 1 + letters.size) % letters.size)
+          }
+        },
+        modifier = Modifier.align(Alignment.CenterStart).testTag("LetterDictionaryBack")) {
+          Icon(
+              Icons.AutoMirrored.Outlined.ArrowBack,
+              tint = MaterialTheme.colorScheme.primary,
+              contentDescription = "Back")
+        }
+
+    // Horizontal Pager for letters in the center
+    HorizontalPager(
+        beyondViewportPageCount = 1,
+        state = pagerState,
+        modifier = Modifier.size(300.dp, 50.dp).align(Alignment.Center).testTag("LetterPager"),
+    ) { page ->
+      val currentLetter = letters[page]
+      Log.d("LetterDictionary", "Current Letter: $currentLetter")
+      Box(
+          contentAlignment = Alignment.Center,
+          modifier = Modifier.fillMaxSize().size(100.dp, 50.dp)) {
+            Box(
+                contentAlignment = Alignment.Center,
+                modifier =
+                    Modifier.border(
+                            2.dp, MaterialTheme.colorScheme.outline, RoundedCornerShape(8.dp))
+                        .background(MaterialTheme.colorScheme.primary, RoundedCornerShape(8.dp))
+                        .clickable { onClick(page, numbOfHeaders) }
```

Comment:
You can add a boolean that can be taken for the parameters of `LetterDictionary` and simple check if it is a clickable or not.
Something like this : 
```
val letterBoxModifier = Modifier
            .border(2.dp, MaterialTheme.colorScheme.outline, RoundedCornerShape(8.dp))
            .background(MaterialTheme.colorScheme.primary, RoundedCornerShape(8.dp))
            .then(
                if (clickable) Modifier.clickable { onClick(page, numbOfHeaders) }
                else Modifier
            )
            .testTag("LetterBox_${currentLetter.uppercaseChar()}")
      Box(
          contentAlignment = Alignment.Center,
          modifier = Modifier.fillMaxSize().size(100.dp, 50.dp)) {
            Box(
                contentAlignment = Alignment.Center,
                modifier =
                letterBoxModifier
                        .testTag("LetterBox_${currentLetter.uppercaseChar()}")) {...} 
```

Reply:
where clickable is the parameter to be given by the `LetterDictionary` that is set to `true `or `false `by default

Reply:
I put your recommendation directly inside the modifier.